### PR TITLE
Introduces Roles

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### Description of the Change
Introduces Roles, which are tags for a group of permission.
The logic is that you can grant a user a role only if you can grant each of the permissions that it comprises of.

### Benefits
Permissioned networks will benefit from native support of roles in Iroha2.

### Possible Drawbacks 
Conditional compilation might introduce bugs which are difficult to detect.

### Jira Issue
https://jira.hyperledger.org/browse/IR-1023

### CODE REVIEW

File: iroha_data_model/src/lib.rs [hunk 1]
```diff
@@ -180,6 +186,19 @@ impl Value {
...
     }
+
+    /// Number of underneath expressions.
+    #[cfg(feature = "roles")]
+    pub fn len(&self) -> usize {
+        use Value::*;
+
+        match self {
+            U32(_) | Id(_) | PublicKey(_) | Bool(_) | Parameter(_) | Identifiable(_)
+            | String(_) | TransactionValue(_) | PermissionToken(_) | Role(_) => 1,
```
Comment (i1i1) [inline iroha_data_model/src/lib.rs:197]: I guess you can put `#[cfg(...)]` on variant itself, am i wrong?

Comment (i1i1) [reply iroha_data_model/src/lib.rs:197]: Just seems like a code duplication to me

Comment (eadventurous) [reply iroha_data_model/src/lib.rs:197]: I think this is not possible. [Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=061996ac7cf9b236d0b9b28f8c61fde7).

Comment (i1i1) [reply iroha_data_model/src/lib.rs:197]: It works this way:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d599bba0120a3ef44f5222d2361509e7

File: iroha/src/tx.rs [hunk 2]
```diff
@@ -219,11 +221,29 @@ impl AcceptedTransaction {
...
+                                .map_err(|reason| NotPermittedFail { reason })
+                                .map_err(TransactionRejectionReason::NotPermitted)?;
+                        }
+                    }
+                } else {
+                    permissions_validator
+                        .check_instruction(&account_id, instruction, world_state_view)
+                        .map_err(|reason| NotPermittedFail { reason })
+                        .map_err(TransactionRejectionReason::NotPermitted)?;
+                }
```
Comment (i1i1) [inline iroha/src/tx.rs:246]: Ew. May be can be substituted with:
```
#[cfg(feature = "roles")]
{
...
}
#[cfg(not(feature = "roles"))]
{
...
}

**Review by i1i1 — APPROVED** [review]

**Review by rkharisov — DISMISSED** [review]

**Review by rkharisov — APPROVED** [review]

**Review by i1i1 — APPROVED** [review]

**Pull request merged**